middleware/logger.go

```diff
@@ -37,59 +58,64 @@ func Logger(next http.Handler) http.Handler {
 	return http.HandlerFunc(fn)
 }
 
-func requestPrefix(reqID string, r *http.Request) *bytes.Buffer {
-	buf := &bytes.Buffer{}
+// FormatRequest stores info from the request, for use by Log()
+func (l *DefaultLogFormatter) FormatRequest(r *http.Request) LogFormatter {
+	var ctx DefaultLogFormatter
```

Comment:
why do you call it `ctx`? that's rather confusing, since `ctx := r.Context()` is commonly used for request context